app/controllers/peer_reviews_controller.rb

```diff
@@ -1,67 +1,167 @@
 class PeerReviewsController < ApplicationController
+  include GroupsHelper
+
   before_action :set_peer_review, only: [:show, :edit, :update, :destroy]
   before_filter :authorize_only_for_admin
 
-  # GET /peer_reviews
-  # GET /peer_reviews.json
   def index
-    @peer_reviews = PeerReview.all
+    @assignment = Assignment.find(params[:assignment_id])
+    @section_column = ''
+    if Section.all.size > 0
+      @section_column = "{
+        id: 'section',
+        content: '" + I18n.t(:'summaries_index.section') + "',
+        sortable: true
+      },"
+    end
+
+    if @assignment.marking_scheme_type == 'rubric'
+      @criteria = @assignment.rubric_criteria
+    else
+      @criteria = @assignment.flexible_criteria
+    end
   end
 
-  # GET /peer_reviews/1
-  # GET /peer_reviews/1.json
-  def show
+  # TODO - Calls the groups table WAY too much... needs fixing
+  def populate
+    @assignment = Assignment.find(params[:assignment_id])  # Needed for the following functions.
+    reviewer_groups = get_groupings_table_info()
+    reviewee_groups = get_groupings_table_info(@assignment.parent_assignment)
+    reviewee_to_reviewers_map = create_map_reviewee_to_reviewers(reviewer_groups, reviewee_groups)
+    id_to_group_names_map = create_map_group_id_to_name(reviewer_groups, reviewee_groups)
+    num_reviews_map = create_map_number_of_reviews_for_reviewer(reviewee_to_reviewers_map)
+    render json: [reviewer_groups, reviewee_groups, reviewee_to_reviewers_map,
+                  id_to_group_names_map, num_reviews_map]
   end
 
-  # GET /peer_reviews/new
-  def new
-    @peer_review = PeerReview.new
+  # Returns a dict of: reviewee_id => [list of reviewers].
+  def create_map_reviewee_to_reviewers(reviewer_groups, reviewee_groups)
+    reviewer_ids = []
+    reviewer_groups.each { |reviewer| reviewer_ids.push(reviewer['id']) }
```

Comment:
Use a `map` instead. You can also do `reviewer.id`, or, better, `&:id` if you use map